pkg/kubelet/cm/dra/manager.go

```diff
@@ -302,38 +335,61 @@ func (m *ManagerImpl) UnprepareResources(pod *v1.Pod) error {
 				pluginName = claimInfo.DriverName
 			}
 
-			// Call NodeUnprepareResource RPC for each resourceHandle
-			client, err := dra.NewDRAPluginClient(pluginName)
-			if err != nil {
-				return fmt.Errorf("failed to get DRA Plugin client for plugin name %s, err=%+v", pluginName, err)
-			}
-			response, err := client.NodeUnprepareResource(
-				context.Background(),
-				claimInfo.Namespace,
-				claimInfo.ClaimUID,
-				claimInfo.ClaimName,
-				resourceHandle.Data)
-			if err != nil {
-				return fmt.Errorf(
-					"NodeUnprepareResource failed, pod: %s, claim UID: %s, claim name: %s, resource handle: %s, err: %+v",
-					pod.Name, claimInfo.ClaimUID, claimInfo.ClaimName, resourceHandle.Data, err)
+			claim := &drapb.Claim{
+				Namespace:      resourceClaim.Namespace,
+				Uid:            string(resourceClaim.UID),
+				Name:           resourceClaim.Name,
+				ResourceHandle: resourceHandle.Data,
 			}
-			klog.V(3).InfoS("NodeUnprepareResource succeeded", "response", response)
+			batches[pluginName] = append(batches[pluginName], claim)
 		}
+		claimInfos[resourceClaim.UID] = claimInfo
+	}
```

Comment:
The comment that begins this loop is now incorrect as we don't call `NodeUnprepareResource()` from within this loop anymore. That comment should be moved below to the new loop with:
```
for pluginName, claims := range batches {
```

Reply:
The "only for the last pod that references the claim" seemed redundant to me. I dropped it in the updated comments.